app/controllers/projects_controller.rb

```diff
@@ -49,6 +49,24 @@ def destroy
     end
   end
 
+  def archive
+    archive_project = @project.project_users.find_by(user_id: current_user)
+    archive_project.archived = true
+    archive_project.save
+    redirect_as_success(edit_projects_path, "project archived")
+  end
+
+  def archived
+    @projects = current_user.project_users.is_archived.map { |project| Project.find(project.project_id) }
```

Comment:
I like naming `archived` to replace `is_archived`
also map code block as mentioned